src/tests/StarGuardJobIntegration.t.sol

```diff
@@ -0,0 +1,300 @@
+        assertEq(args, "No distribution", "unexpected calldata with no spell");
+    }
+
+    function testWorkWithDelayedSpell() public {
+        // Prepare job
+        job.add(starGuardSpark);
+
+        // Prepare delayed spell
+        uint256 executableAt = block.timestamp + 1 days;
+        address delayedStarSpell = address(new DelayedStarSpell(executableAt));
+
+        // Prepare StarGuard
+        vm.startPrank(pauseProxy);
+        StarGuardLike(starGuardSpark).file("maxDelay", type(uint160).max);
+        StarGuardLike(starGuardSpark).plot(delayedStarSpell, delayedStarSpell.codehash);
+        vm.stopPrank();
+
+        // Check state immidiately
+        {
+            (bool canWork,) = job.workable(NET_A);
+            assertFalse(canWork, "unexpected workable() true with delayed spell: immidiate");
+        }
+
+        // Check state at executableAt - 1
+        {
+            vm.warp(executableAt - 1);
+            (bool canWork,) = job.workable(NET_A);
+            assertFalse(canWork, "unexpected workable() true with delayed spell: almost at executable");
+        }
+
+        // Check state at executableAt
+        {
+            vm.warp(executableAt);
+            uint256 beforeWorkable = vm.snapshot();
+            (bool canWork, bytes memory args) = job.workable(NET_A);
+            assertTrue(canWork, "unexpected workable() false with delayed spell: at executable");
+            
+            // Work
+            vm.revertTo(beforeWorkable); // snapshot is required as `workable` modifies state
+            job.work(NET_A, args);
+        }
+    }
+
+    function testWorkableWithInvalidSpell() public {
+        // Prepare job
+        job.add(starGuardSpark);
+
+        // Prepare malicious spell
+        address maliciousStarSpell = address(new InvalidStarSpell());
+
+        // Prepare StarGuard
+        vm.prank(pauseProxy);
+        StarGuardLike(starGuardSpark).plot(maliciousStarSpell, maliciousStarSpell.codehash);
+
+        // Ensure that the `prob` returns true, so the spell is technically ready to be executed
+        assertTrue(StarGuardLike(starGuardSpark).prob(), "unexpected `prob` value after `plot`");
+
+        // Check workable state
+        (bool canWork,) = job.workable(NET_A);
+        assertFalse(canWork, "unexpected workable() true with malicious spell");
+    }
+
+    function testWorkableWithMaliciousSpell() public {
+        // Prepare job
+        job.add(starGuardSpark);
+
+        // Prepare malicious spell
+        address maliciousStarSpell = address(new MaliciousStarSpell(starGuardSpark));
+
+        // Prepare StarGuard
+        vm.prank(pauseProxy);
+        StarGuardLike(starGuardSpark).plot(maliciousStarSpell, maliciousStarSpell.codehash);
+
+        // Ensure that the `prob` returns true, so the spell is technically ready to be executed
+        assertTrue(StarGuardLike(starGuardSpark).prob(), "unexpected `prob` value after `plot`");
```

Comment:
This check is superfluous if redefining the implementation of `workable()` in terms try to call `work()`.